In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.tree import DecisionTreeClassifier, DecisionTreeRegressor

In [3]:
data = pd.read_csv(r"E:\Nextstep\Scaler\Business-Case-Study\Ola\Dataset\ola_driver_scaler.csv")
data.head()

Unnamed: 0    MMM-YY  Driver_ID   Age  Gender City  Education_Level  \
0           0  01/01/19          1  28.0     0.0  C23                2   
1           1  02/01/19          1  28.0     0.0  C23                2   
2           2  03/01/19          1  28.0     0.0  C23                2   
3           3  11/01/20          2  31.0     0.0   C7                2   
4           4  12/01/20          2  31.0     0.0   C7                2   

   Income Dateofjoining LastWorkingDate  Joining Designation  Grade  \
0   57387      24/12/18             NaN                    1      1   
1   57387      24/12/18             NaN                    1      1   
2   57387      24/12/18        03/11/19                    1      1   
3   67016      11/06/20             NaN                    2      2   
4   67016      11/06/20             NaN                    2      2   

   Total Business Value  Quarterly Rating  
0               2381060                 2  
1               -665480                 2  
2                     0                 2  
3                     0                 1  
4                     0                 1

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19104 entries, 0 to 19103
Data columns (total 14 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            19104 non-null  int64  
 1   MMM-YY                19104 non-null  object 
 2   Driver_ID             19104 non-null  int64  
 3   Age                   19043 non-null  float64
 4   Gender                19052 non-null  float64
 5   City                  19104 non-null  object 
 6   Education_Level       19104 non-null  int64  
 7   Income                19104 non-null  int64  
 8   Dateofjoining         19104 non-null  object 
 9   LastWorkingDate       1616 non-null   object 
 10  Joining Designation   19104 non-null  int64  
 11  Grade                 19104 non-null  int64  
 12  Total Business Value  19104 non-null  int64  
 13  Quarterly Rating      19104 non-null  int64  
dtypes: float64(2), int64(8), object(4)
memory usage: 2.0+ MB


In [5]:
data.describe()

Unnamed: 0     Driver_ID           Age        Gender  \
count  19104.000000  19104.000000  19043.000000  19052.000000   
mean    9551.500000   1415.591133     34.668435      0.418749   
std     5514.994107    810.705321      6.257912      0.493367   
min        0.000000      1.000000     21.000000      0.000000   
25%     4775.750000    710.000000     30.000000      0.000000   
50%     9551.500000   1417.000000     34.000000      0.000000   
75%    14327.250000   2137.000000     39.000000      1.000000   
max    19103.000000   2788.000000     58.000000      1.000000   

       Education_Level         Income  Joining Designation         Grade  \
count     19104.000000   19104.000000         19104.000000  19104.000000   
mean          1.021671   65652.025126             1.690536      2.252670   
std           0.800167   30914.515344             0.836984      1.026512   
min           0.000000   10747.000000             1.000000      1.000000   
25%           0.000000   42383.000000             1.000000      1.000000   
50%           1.000000   60087.000000             1.000000      2.000000   
75%           2.000000   83969.000000             2.000000      3.000000   
max           2.000000  188418.000000             5.000000      5.000000   

       Total Business Value  Quarterly Rating  
count          1.910400e+04      19104.000000  
mean           5.716621e+05          2.008899  
std            1.128312e+06          1.009832  
min           -6.000000e+06          1.000000  
25%            0.000000e+00          1.000000  
50%            2.500000e+05          2.000000  
75%            6.997000e+05          3.000000  
max            3.374772e+07          4.000000

In [6]:
data.describe(include='object')

MMM-YY   City Dateofjoining LastWorkingDate
count      19104  19104         19104            1616
unique        24     29           869             493
top     01/01/19    C20      23/07/15        29/07/20
freq        1022   1008           192              70

## Unique Values

In [7]:
data.nunique()

Unnamed: 0              19104
MMM-YY                     24
Driver_ID                2381
Age                        36
Gender                      2
City                       29
Education_Level             3
Income                   2383
Dateofjoining             869
LastWorkingDate           493
Joining Designation         5
Grade                       5
Total Business Value    10181
Quarterly Rating            4
dtype: int64

In [8]:
for col in data.columns:
    if data[col].nunique() < 6:
        print(col, ": ",  data[col].unique())

Gender :  [ 0.  1. nan]
Education_Level :  [2 0 1]
Joining Designation :  [1 2 3 4 5]
Grade :  [1 2 3 4 5]
Quarterly Rating :  [2 1 4 3]


#### Inference:
- Unnamed: 0 contains unique values, doesn't helpful for EDA and model building
- 24 reporting days of drivers are collected in the dataset
- Total number of drivers in the dataset are 2381
- Dataset Characteristics
    - No of drivers: 2381
    - No. of Cities: 29
    - Joining designation are sectored into 5 categories (1 to 5)
    - 5 different grades are given to the drivers (1 to 5)
    - Quarterly ratings are grouped as 4 categories (1 to 4)

## Null values

In [9]:
data.isnull().sum()

Unnamed: 0                  0
MMM-YY                      0
Driver_ID                   0
Age                        61
Gender                     52
City                        0
Education_Level             0
Income                      0
Dateofjoining               0
LastWorkingDate         17488
Joining Designation         0
Grade                       0
Total Business Value        0
Quarterly Rating            0
dtype: int64

In [10]:
data['LWD'] = data['LastWorkingDate'].fillna(1)
data.groupby('Driver_ID')['LWD']

0               1
1               1
2        03/11/19
3               1
4               1
           ...   
19099           1
19100           1
19101           1
19102           1
19103           1
Name: LastWorkingDate, Length: 19104, dtype: object